BrowserStack.java

```diff
@@ -37,32 +38,113 @@ public static void myassert(boolean  x) {
     //push() - pushes the element in this case its a url (String)
     private  void push(String url){
         //WRITE YOUR CODE HERE
+    	
+    	top ++;
+    	myStack.add(top, url);
+    	    	
+    		
     }
     //pop()- pops the element from the stack
     private String pop(){
         //WRITE YOUR CODE HERE
-
+    	if (!isEmpty()){
+		 String popUrl = myStack.remove(top);
+		 top--;
+		 // System.out.println("pop" +popUrl);
+		 	return popUrl;
+    	}
+    	else
+    		return("Stack is empty");
     }
 
-    private void printStack(){
+   private void printStack(){
         //WRITE YOUR CODE HERE
+	   if (!isEmpty()){
+	    String popValue=pop();
+	     System.out.println("element=" + popValue);
+	     if(isEmpty())
+	     {  
+	    	 push(popValue); //break the recursive when the stack is empty
+	    	 return;
+	     }
+	     printStack(); // using recursive method to print the stack
+	     
+	     push(popValue);
+	     
+	   }      
 
     }
     private  String peek(){
         //WRITE YOUR CODE HERE
+    	 System.out.println("peekelement=" +myStack.get(top));
+         return( myStack.get(top));
+         
 
     }
-    private  int size(){
+   private  int size(){
         //WRITE YOUR CODE HERE
+         return(top+1);
 
     }
     private  boolean isEmpty(){
         //WRITE YOUR CODE HERE
-
+		if(top !=-1)
+			return false;
+		else 
+			return true;
+		
+		
     }
-    private  boolean search(String url){
-        //WRITE YOUR CODE HERE
+  /* private void findBrowser(String url) {  
+    										// Searching the element in the stack using recursive method
+	   
+	   String popValue = pop();
+	   if (url.equals(popValue) && !isEmpty()){
+		   push(popValue);						// if the search value is found or stack is empty break the recursive loop
+		   isUrlFound = true;
+		   return;
+	   }
+	   else if(isEmpty()){
+		   push(popValue);
+		   isUrlFound = false;
+		   return;
+	   }
+		   findBrowser(url);  //
+		   push(popValue);
+   }*/
+    
+   private  boolean search(String url){
+      
+	   //WRITE YOUR CODE HERE
+	/*   findBrowser(url);     // Searching the element in the stack using recursive method
+	   return isUrlFound;*/
+
 
+	   String[] newStackArr = new String[top+1]; // initializing temporary array to store the popped value
+	   String urlPop="";
```

Comment:
**Naming convention**  Java 👮 🚓 : Try to avoid giving method names for variables. urlPop